tools/lib/tests/test_caching.py

```diff
@@ -34,6 +35,28 @@ def do_HEAD(self):
 
 class TestFileDownload(unittest.TestCase):
 
+  def test_defaults(self):
+    # TODO: parameterize the defaults so we don't rely on hard-coded values in xx
+
+    self.assertEqual(URLFile.pool_manager().pools._maxsize, 10) # PoolManager num_pools param
+    pool_manager_defaults = {
+      "maxsize": 100,
+      "socket_options": [(socket.SOL_SOCKET, socket.SO_KEEPALIVE, 1),],
+    }
+    for k, v in pool_manager_defaults.items():
+      self.assertEqual(URLFile.pool_manager().connection_pool_kw.get(k), v)
+
+    retry_defaults = {
+      "total": 5,
+      "backoff_factor": 0.5,
+      "status_forcelist": [409, 429, 503, 504],
+    }
+    for k, v in retry_defaults.items():
+      self.assertEqual(getattr(URLFile.pool_manager().connection_pool_kw["retries"], k), v)
+
+     # caching off by default
+    self.assertEqual(URLFile("http://fake.url")._force_download, True)
```

Comment:
this is both a brittle test and doesn't directly test what you want. checking that the file doesn't end up in `Paths.download_cache_root()` would be a better test, but ideally we check if anything was written to disk